### PR TITLE
chore: refine gene info searches

### PR DESCRIPTION
**Pull request by sienacizdziel**
- #TICKET_NUMBER

### Reviewers
**Functional:** 
@MDunitz 
**Readability:** 

---

## Changes
- refine ncbi api searches with additional parameters
- fix 'zuckerberg' typo in endpoint yml files
- modify api tests to incorporate refined api searches
- add rough python script for gene info edge casing testing into /scripts
- replace is_ensembl_id_result in response body with show_warning_banner

## Notes for Reviewer
Per comments from Ambrose, Arathi, and Pablo during my gene info presentation, this code refines NCBI API searches to also include organism and gene name labels, which increases successful gene info searches. I also went ahead and added my script for my edge case testing here: [counts](https://docs.google.com/spreadsheets/d/1ZULeY_rBsN8u4sUREmJZC3P3PL2yZm2pPrRvSCpCMHQ/edit#gid=528908839)

### CODE REVIEW

File: scripts/gene_info_hits.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# This script was created to test edge cases of the gene_info API endpoint.
```
Comment (seve) [inline scripts/gene_info_hits.py:1]: 🙌

File: backend/config/corpora-api.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 openapi: 3.0.0
 info:
   version: "1.0.0"
-  title: Chan Zuckerburg Initiative cellxgene Data Portal API
+  title: Chan Zuckerberg Initiative cellxgene Data Portal API
```
Comment (danieljhegeman) [inline backend/config/corpora-api.yml:4]: 🤦‍♂️

File: scripts/gene_info_hits.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+for row in data:
+    url = f'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=gene&term=${row[0]}&retmode=json'
+    result = json.loads(urllib.request.urlopen(url).read())
+
+    # unsuccessful search result (success is when the code returns one UID from NCBI)
+    if (
+        "esearchresult" not in result or 
+        "idlist" not in result["esearchresult"] or 
+        len(result["esearchresult"]["idlist"]) < 1
+        ):
```
Comment (danieljhegeman) [inline scripts/gene_info_hits.py:25]: nit: move `):` back one tab

Comment (danieljhegeman) [reply scripts/gene_info_hits.py:25]:
```suggestion
    ):
```

File: scripts/gene_info_hits.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        print(result)
+        if len(result["esearchresult"]["idlist"]) == 1:
+            writer.writerow(['SUCCESS', row[0], row[1], len(result["esearchresult"]["idlist"]), result])
+
+        # too many ID results
+        else:
+            writer.writerow(['TOO MANY RESULTS', row[0], row[1], len(result["esearchresult"]["idlist"]), result])
+
+file.close()
+counts.close()
```
Comment (danieljhegeman) [inline scripts/gene_info_hits.py:46]: add newline

File: backend/config/corpora-api.yml (hunk 2 above)
Comment (MDunitz) [reply backend/config/corpora-api.yml:4]: 🤣

File: backend/gene_info/api/ncbi_provider.py [hunk 5]
```diff
@@ -62,11 +62,15 @@ def fetch_gene_uid(self, geneID, gene):
             raise NCBIUnexpectedResultException
 
         # search with gene name if needed
-        if not self._is_valid_search_result(search_response) and gene != "":
+        if not self._is_valid_search_result(search_response) and gene and gene != "":
             try:
-                search_response = self._search_gene_uid(gene)
+                term = "(" + str(gene) + "%5BGene%20Name%5D)%20AND%20human%5BOrganism%5D"
```
Comment (MDunitz) [inline backend/gene_info/api/ncbi_provider.py:67]: can you pull `%5BGene%20Name%5D)%20AND%20human%5BOrganism%5D` out as a const and reuse it in the tests?

File: backend/gene_info/api/ncbi_provider.py [hunk 6]
```diff
@@ -62,11 +62,15 @@ def fetch_gene_uid(self, geneID, gene):
...
+                search_response = self._search_gene_uid(term)
                 if self._is_valid_search_result(search_response):
-                    return (int(search_response["esearchresult"]["idlist"][0]), False)
+                    # switched is_ensembl_id_result to True with new search term
```
Comment (MDunitz) [inline backend/gene_info/api/ncbi_provider.py:70]: this comment wont make much sense outside of the pr review (but I do appreciate it)

could you also add a comment to explaining the string on line 67 (or pull it out into a const with a clear name)?

**Review by MDunitz — APPROVED** [review]
lgtm -- thanks for making all the updates for readability!
